pkg/messaging/direct_messaging.go

```diff
@@ -566,7 +566,7 @@ func (d *directMessaging) getRemoteApp(appID string) (remoteApp, error) {
 		address = d.checkHTTPEndpoints(id)
 	} else {
 		request := nr.ResolveRequest{ID: id, Namespace: namespace, Port: d.grpcPort}
-		address, err = d.resolver.ResolveID(request)
+		address, err = d.resolver.ResolveID(context.TODO(), request)
```

Comment:
Can we update `getRemoteApp` to accept a context?

Reply:
I was thinking this would be a separate PR (hence the TODO).